Studio-Lovelies/GG-JointJustice-Unity
# Refactor/simplified object inventory

## Summary

A refactor which aims to simplify the previous object inventory system by using a list to store items currently in the court record, instead of an ordered dictionary. Functionality should remain the same.
Fixes actors and evidence sometimes being pulled from OrderedObjectStorage instead of the dictionary of available objects (hence why an ordered dictionary was no longer required).

OrderedObjectStorage is replaced with ObjectLookup which is used for getting objects from the dictionary of available objects and storing them in a list so they can be displayed on the evidence/profiles menu.
Makes ObjectInventory class a lot more "humble", moving logic to ObjectLookup for testing.
Replaces OrderedObjectStorage tests with ObjectLookup tests.
Renames some folders and game objects to remove the word "Dictionary".

## Before/after screenshots and/or animated gif

No visual changes. Tests pass.
<img width="331" alt="Screenshot 2021-10-03 at 23 28 58" src="https://user-images.githubusercontent.com/67964179/135773477-b288ed80-a87c-483d-a94a-3ea38a27636b.png">

## Testing instructions

Run the PlayMode and EditMode tests. Play ShowItemDemo and EvidenceAddingTestScript narrative scripts and see if they function as expected. For ShowItemDemo the backgrounds are broken, but the items appearing should work.

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:
